aleph0/createActions.ts

```diff
@@ -0,0 +1,86 @@
+import { RunnableFunctionWithParse } from "openai/lib/RunnableFunction";
+import { z } from "zod";
+import { loadSnippets } from "./utils";
+import { readFile } from "fs";
+import { TechnologySnippets } from "./types";
+
+export const createActions = (): Record<
+  string,
+  RunnableFunctionWithParse<any>
+> => {
+  return {
+    // createTaskDescription: {
+    //   function : async (args: { userPrompt: string }) => {
+    //     return {
+    //       taskDescription: await createTaskDescriptionPrompt(args.userPrompt), //TODO: need to call open AI here
+    //     };
+    //   },
+    //   name: "createTaskDescription",
+    //   description: "Returns a more complete task description before applying actions.",
+    //   parse: (args: string) => {
+    //     return z
+    //       .object({
+    //         userPrompt: z.string(),
+    //       })
+    //       .parse(JSON.parse(args));
+    //   },
+    //   parameters: {
+    //     type: "object",
+    //     properties: {
+    //       userPrompt: {
+    //         type: "string",
+    //       },
+    //     },
+    //   },
+    // },
+    getSnippets: {
+      function: async(args: {technology: string}) => {
+        return { snippets: await loadSnippets(args.technology) };
+      },
+      name: "getSnippets",
+      description: "Returns the snippets for a given technology (next13...).",
+      parse: (args: string) => {
+        return z
+          .object({
+            technology: z.string(),
+          })
+          .parse(JSON.parse(args));
+      },
+      parameters: {
+        type: "object",
+        properties: {
+          technology: {
+            type: "string",
+          },
+        },
+      },
+    }
+    readSnippet: {
+      function: async(args: {snippetName: string, snippets: TechnologySnippets}) => {
```

Comment:
`snippets: TechnologySnippets` - not sure you need that. We decide where OpenAI reads from. AI doesn't need to pass this to us.

Reply:
You mean it remembers the output of the `getSnippets` and can utalize that? Should we then use absolute paths in everything?